### PR TITLE
♻️ [youtube/instagram] Split component files for packaging

### PR DESCRIPTION
**Pull request by rcebulko**
I think `amp-video` is already split appropriately (although without the `component` naming convention, as it may not be appropriate in that case) so the dependency on `VideoBaseElement` should still be pure enough for Preact packaging

### CODE REVIEW

File: extensions/amp-instagram/1.0/base-element.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+import {Instagram} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {dict} from '../../../src/utils/object';
+
+export class BaseElement extends PreactBaseElement {
+  /** @override */
+  init() {
+    return dict({
```
Comment (rcebulko) [inline extensions/amp-instagram/1.0/base-element.js:24]: The more I look at this the more it feels like it belongs on the AMP side

File: extensions/amp-instagram/1.0/amp-instagram.js [hunk 2]
```diff
@@ -34,36 +32,8 @@ class AmpInstagram extends PreactBaseElement {
     );
     return super.isLayoutSupported(layout);
   }
-
-  /** @override */
-  init() {
-    return dict({
-      'requestResize': (height) => {
-        this.forceChangeHeight(height);
```
Comment (jridgewell) [inline extensions/amp-instagram/1.0/amp-instagram.js:42]: `forceChangeHeight` is an AMP method, so this should remain here.

File: extensions/amp-instagram/1.0/base-element.js (hunk 1 above)
Comment (jridgewell) [reply extensions/amp-instagram/1.0/base-element.js:24]: Yup

File: extensions/amp-instagram/1.0/amp-instagram.js (hunk 2 above)
Comment (rcebulko) [reply extensions/amp-instagram/1.0/amp-instagram.js]: Done.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**